### PR TITLE
[201911] Adding SKU Mellanox-SN3800-D100C12S2

### PR DESCRIPTION
**Pull request by madhanmellanox**
#### Why I did it
To create a new SKU Mellanox-SN3800-D100C12S2

#### How I did it
I arrived at the SKU configuration values based on the following SKU template, Port mapping and number of uplinks and downlinks.

SKU template:
Port configuration
• Breakout mode for each port - Defined in port mapping
• Speed of the port Defined in Port mapping
• Auto-negotiation enable/disable No setting required
• FEC mode No setting required
• Type of transceiver used Not needed
Buffer configuration
• Shared headroom enable
• If shared headroom enabled what is the over-subscription ratio as in SN3800
• Dynamic Buffer disable
• In static buffer scenario how many uplinks and downlinks? as in SN3800
• 2km cable support required? no
Switch configuration
• Warmboot enabled? yes
• Should warmboot be added to SAI profile when enabled? yes
• Is VxLAN source port range set? yes
• Should Vxlan source port range be added to SAI profile when set. as in SN3800
• Is Static Policy Based Hashing enabled? no

Port Mapping
etp1 to etp36 split into 50G
etp37 and etp40 is 10G
etp38 and etp39 splint into 50G
etp41 to etp52 split into 50G
etp53 to etp64 is 100G

Number of Uplinks / Downlinks:
TO topology: 12 100G uplinks and rest all downlinks.
T1 topology: (SKU will not be used in T1 topology), so same 12 100G uplinks and rest all downlinks is used to arrive at buffer config values.

#### How to verify it
Build the image, install it on the 3800 switch and set the SKU and verify the ports come up with proper speeds.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog

Changes are in sonic-buildimage/device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/ folder.

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/sai.profile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/sai_3800_2x10g_28x50g_48x100g.xml
```
Comment (prsunny) [inline device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/sai.profile:1]: why is the xml file different from` sai_3800_2x10g_100x50g_12x100g.xml`?

**Comment by prsunny** [comment]
Can someone from NVDIA team sign-off? @liat-grozovik , @volodymyrsamotiy .

File: device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/sai.profile (hunk 1 above)
Comment (prsunny) [reply device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/sai.profile:1]: How is this change tested?

Comment (madhanmellanox) [reply device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/sai.profile:1]: okay, this has to be fixed. I will fix it.

File: device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/port_config.ini [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Ethernet132  132,133          etp34a        34    50000
+Ethernet134  134,135          etp34b        34    50000
+Ethernet136  136,137          etp35a        35    50000
+Ethernet138  138,139          etp35b        35    50000
+Ethernet140  140,141          etp36a        36    50000
+Ethernet142  142,143          etp36b        36    50000
+Ethernet144  144,145,146,147  etp37         37    10000
+Ethernet148  148,149          etp38a        38    50000
+Ethernet150  150,151          etp38b        38    50000
+Ethernet152  152,153          etp39a        38    50000
```
Comment (prsunny) [inline device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/port_config.ini:77]: isn't this 39 (index) ?

File: device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/qos.json.j2 [hunk 3]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/qos.json.j2
```
Comment (prsunny) [inline device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/qos.json.j2:1]: pg_profile is pointing to ../Mellanox-SN3800-D112C8/pg_profile_lookup.ini. Can you confirm if these soft-links are correct?

Comment (madhanmellanox) [reply device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/qos.json.j2:1]: yes I verified it on the workspace, the symbolic link is there. I also verified the pg_profile_lookup.ini file is viewable on the switch as well.

File: device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/port_config.ini (hunk 2 above)
Comment (madhanmellanox) [reply device/mellanox/x86_64-mlnx_msn3800-r0/Mellanox-SN3800-D100C12S2/port_config.ini:77]: yes, it is. Sorry, I will fix it.

**Review by robocoder99 — APPROVED** [review]
Approve with exception of 10G ports needing validation.

**Comment by prsunny** [comment]
@liat-grozovik , @volodymyrsamotiy , @dgsudharsan , can we merge this?

**Review by dgsudharsan — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**